Feature: Silence all notifiers

xiaozhikang0916 opened the pull request:
# Silencio
When playing games in full screen mode or watching video, some may feel disturbed if the Notifier pop up a window and then checkout to desktop.
Add a checkbox in Setting/Notifier to disable(checked)/enable(unchecked) all notifier.
![qq 20160903115416](https://cloud.githubusercontent.com/assets/8273026/18222549/365d3c86-71cd-11e6-9509-f8ff1ef6651b.png)

* Silencio is a [neta from Harry Potter](https://en.wikipedia.org/wiki/List_of_spells_in_Harry_Potter)
* If the checkbox is checked, it will override all the Enable settings of notifiers to  make it disabled
* However, it will **not** enable those notifiers which are set disabled

## Modified
It seems that I have made a large commit because of the line-ending style, sorry about that.

Here are what i modified in the code:

1. A new property `IsSilenced` in `Notifier/NotifierBase`
2. If-case in `Notifier/NotifierBase#Notify()`
3. New checkboxes in `Window/Dialog/DialogConfiguration` and relative check-event
4. `Window/Dialog/DialogConfiguration#_Constructor()` to initial the checkbox

## Todo
I also want it set silenced automatically when some other games run in full screen mode, but can't find a way to listen full-screen event yet, so another checkbox in DialogConfiguration is set invisible for future develop.
Do you have any idea?

andanteyk commented:
Thanks for your cooperation.
I make adjustments in the commit 64e1c166ed1f8a59191abfcf7930d4f0c048beca.

Details:

* Translation into Japanese.
Sorry, I changed the text in the Configuration window to the general sentence.
"Silencio" is assumed to be a minor spell; It might be difficult to understand what to use.
I've left in the tooltip instead. Please pardon.

* Setting-related adjustment.
From UI event of setting window, it should not rewrite settings directly. It will also be reflected as pressing "Cancel" button.
Therefore, I fixed them.

* Formatting.

xiaozhikang0916 commented:
Glad to find that you have accepted my code, i'm ok with the adjustment.
And, it's also OK to write in Japanese ( read your complaint from your
twitter)
By the way, do you like to maintain a TODO list of the project, so that
someone else can contribute to it in the spare time?

Andante <notifications@github.com>于2016年9月5日周一 上午1:03写道：

> Thanks for your cooperation.
> I make adjustments in the commit 64e1c16
> <https://github.com/andanteyk/ElectronicObserver/commit/64e1c166ed1f8a59191abfcf7930d4f0c048beca>
> .
>
> Details:
>
>    -
>
>    Translation into Japanese.
>    Sorry, I changed the text in the Configuration window to the general
>    sentence.
>    "Silencio" is assumed to be a minor spell; It might be difficult to
>    understand what to use.
>    I've left in the tooltip instead. Please pardon.
>    -
>
>    Setting-related adjustment.
>    From UI event of setting window, it should not rewrite settings
>    directly. It will also be reflected as pressing "Cancel" button.
>    Therefore, I fixed them.
>    -
>
>    Formatting.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/andanteyk/ElectronicObserver/pull/72#issuecomment-244614822>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH48goRUBLYyCGlUwgGBwvcrJWKYqB9Lks5qmvnsgaJpZM4J0N3E>
> .
>

andanteyk commented:
了解しました。
あまり利用していませんが、 github の Issue がそれに当たるかと思います。
enhancement label をつけておくとよいかもしれません。
